src/couch/src/couch_httpd_auth.erl

```diff
@@ -460,16 +460,18 @@ cookie_auth_header(
 cookie_auth_header(_Req, _Headers) ->
     [].
 
-cookie_auth_cookie(Req, User, Secret, TimeStamp, MustMatchBasic) ->
-    MustMatchBasicStr =
-        case MustMatchBasic of
-            true -> "1";
-            false -> "0"
-        end,
-    SessionData = lists:join(":", [
+cookie_auth_cookie(Req, User, Secret, TimeStamp, true) ->
+    SessionItems = [
         User,
-        lists:join(",", [erlang:integer_to_list(TimeStamp, 16), MustMatchBasicStr])
-    ]),
+        [erlang:integer_to_list(TimeStamp, 16), ",1"]
+    ],
+    cookie_auth_cookie(Req, Secret, SessionItems);
+cookie_auth_cookie(Req, User, Secret, TimeStamp, false) ->
+    SessionItems = [User, erlang:integer_to_list(TimeStamp, 16)],
+    cookie_auth_cookie(Req, Secret, SessionItems).
+
+cookie_auth_cookie(Req, Secret, SessionItems) when is_list(SessionItems) ->
+    SessionData = lists:join(":", SessionItems),
```

Comment:
This is a bit tricky to reason about:

```erlang
(node1@127.0.0.1)1> SessionItems = [<<"user">>, ["abcd123", ",1"]].
[<<"user">>,["abcd123",",1"]]
(node1@127.0.0.1)2> SessionData = lists:join(":", SessionItems).
[<<"user">>,":",["abcd123",",1"]]
```

However it does work as https://www.erlang.org/doc/man/crypto#mac-4 takes iodata.

```erlang
(node1@127.0.0.1)3> erlang:iolist_to_binary(SessionData).
<<"user:abcd123,1">>
```

The existing cookie will then be:

```erlang
(node1@127.0.0.1)1> SessionItems = [<<"user">>, ["abcd123"]].
[<<"user">>,["abcd123"]]
(node1@127.0.0.1)2> SessionData = lists:join(":", SessionItems).
[<<"user">>,":",["abcd123"]]
(node1@127.0.0.1)3> erlang:iolist_to_binary(SessionData).
<<"user:abcd123">>
```

Reply:
the intention was to make it less tricky to reason about. :(

the earlier PR exploited the fact that we accept iodata (whereas couchdb before 1.0 perhaps didn't, or we weren't familiar enough with erlang, etc)

Reply:
The new way is more clear. It's obvious it's iodata. Previously `?b2l(User) ++ ":" ++ erlang:integer_to_list(TimeStamp, 16)` hinted that perhaps we had to have a flat list.